packages/nextra/src/mermaid.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import mermaid from 'mermaid'
+
+/**
+ * Assign a unique ID to each mermaid svg as per requirements of `mermaid.render`.
+ */
+let id = 0
+
+export default function Mermaid({ chart }: { chart: string }): ReactElement {
+  const [svg, setSVG] = useState('')
+
+  useEffect(() => {
+    mermaid.initialize({
+      startOnLoad: true,
+      theme:
+        typeof localStorage === 'undefined' ? undefined : localStorage.theme
```

Comment:
any through on how to access `useTheme` hook inside this component and use `theme` value here?

Reply:
I think you can directly use `useTheme` in this component